scripts/pushsafer.pl

```diff
@@ -0,0 +1,318 @@
+}
+
+sub strip_formating {
+    my ($msg) = @_;
+    $msg =~ s/\x03[0-9]{0,2}(,[0-9]{1,2})?//g;
+    $msg =~ s/[^\x20-\xFF]//g;
+    $msg =~ s/\xa0/ /g;
+    return $msg;
+}
+
+
+# check our away status & pushsafer_only_if_away. returns 0 if it's ok to send a message. 
+sub check_away {
+    my ($server) = @_;
+    my $msg_only_if_away = Irssi::settings_get_bool('pushsafer_only_if_away');
+    if ($msg_only_if_away && $server->{usermode_away} != '1') {
+        debug("Only sending messages if we're marked as away, and we're not");
+        return 1;
+    }
+    return 0;
+}
+
+sub check_ignore {
+    return 0 unless(Irssi::settings_get_bool('pushsafer_ignore'));
+    my @ignores = read_file();
+    return 0 unless(@ignores);
+    my ($mask) = @_;
+
+    foreach (@ignores) {
+        $_ =~ s/\./\\./g;
+        $_ =~ s/\*/.*?/g;
+        if ($mask =~ m/^$_$/i) {
+            debug('Ignore matches, not pushing.');
+            return 1;
+        }
+    }
+    return 0;
+}
+
+sub check_ignore_channels {
+    my ($target) = @_;
+    my @ignore_channels = split(' ', Irssi::settings_get_str('pushsafer_ignorechannels'));
+    return 0 unless @ignore_channels;
+    if (grep {lc($_) eq lc($target)} @ignore_channels) {
+        debug("$target set as ignored channel.");
+        return 1;
+    }
+    return 0;
+}
+
+sub ignore_handler {
+    my ($data, $server, $item) = @_;
+    $data =~ s/\s+$//g;
+    Irssi::command_runsub('pushignore', $data, $server, $item);
+}
+
+sub ignore_unknown {
+    cmd_help();
+    Irssi::signal_stop(); # Don't print 'no such command' error.
+}
+
+sub ignore_list {
+    my @data = read_file();
+    if (@data) {
+        my $i = 1;
+        my $out;
+        foreach(@data) {
+            $out .= $i++.". $_\n";
+        }
+        chomp $out;
+        Irssi::print($out, MSGLEVEL_CLIENTCRAP);
+    }
+}
+
+sub ignore_add {
+    my ($data, $server, $item) = @_;
+    $data =~ s/^([\s]+).*$/$1/;
+    return Irssi::print("No hostmask given.", MSGLEVEL_CLIENTCRAP) unless($data ne "");
+
+    my @ignores = read_file();
+    push(@ignores, $data);
+    write_file(@ignores);
+    Irssi::print("Successfully added '$data'.", MSGLEVEL_CLIENTCRAP);
+}
+
+sub ignore_remove {
+    my($num, $server, $item) = @_;
+    $num =~ s/^(\d+).*$/$1/;
+    return Irssi::print("List-number is needed when removing", MSGLEVEL_CLIENTCRAP) unless(looks_like_number($num));
+    my @ignores = read_file();
+    
+    # Index out of range
+    return Irssi::print("Number was out of range.", MSGLEVEL_CLIENTCRAP) unless(scalar(@ignores) >= $num);
+    delete $ignores[$num-1];
+    write_file(@ignores); 
+}
+
+sub write_file {
+    read_settings();
+    my $fp;
+    if (!open($fp, ">", $pushsafer_ignorefile)) {
+        Irssi::print("Error opening ignore file", MSGLEVEL_CLIENTCRAP);
+        return;
+    }
+    print $fp join("\n", @_);
+    close $fp;
+}
+
+sub read_file {
+    read_settings();
+    my $fp;
+    if (!open($fp, "<", $pushsafer_ignorefile)) {
```

Comment:
maybe you should just check if an ignore file exists and return silently if that is not a case, that would safe the user having to "touch" an empty file